### ISSUE TITLE
Doc page for "Object Storage events source" - advocate review issues

### ISSUE THREAD

**Issue by jthomas** [IBM-Bluemix-Docs/openwhisk#120]
I've run through the doc instructions with my "external developer 🎩" on to try to spot issues and corrections needed to improve the developer experience.

Overall, everything worked 👍 but there are a number of things to clarify to help those will less experience with the platform understand what's going on. Feedback is broken down into the doc sections it references. 

## General Issues
- [x] Multiple page links are broken from the docs are 404'ing including...
  - ["Provision an instance Object Storage"](https://cloud.ibm.com/docs/services/cloud-object-storage/basics?topic=cloud-object-storage-for-developers#provision-an-instance-of-ibm-cloud-object-storage)
  - ["IBM Cloud Object Storage documentation"](https://cloud.ibm.com/docs/services/cloud-object-storage?topic=cloud-object-storage-select_endpoints#select_endpoints)

## [Creating a trigger to respond to the changes feed](https://cloud.ibm.com/docs/openwhisk?topic=cloud-functions-cloud_object_storage#creating_a_trigger_cos)

- [x] Trigger create command is missing a backslash and does not work, should be...
```
ibmcloud fn trigger create myCosTrigger --feed myCosPkg/changes \
--param bucket myBucket \
--param endpoint s3.us-south.cloud-object-storage.appdomain.cloud
```

- [x] `ibmcloud fn activation poll` command is not explained. It is just dropped in after the trigger creation. Can we add a step explaining what this command is and why we run it?
- [x] Can we change the demo code below to be a bit more realistic? Make the `console.log` print a bit more of a formatted message and return some key fields rather than a null?
```
function main(data) {
console.log(data);
}
```
- [x] Instructions do not explain the demo code needs to be saved in the `showCosChange.js` file.

## [Data structure for trigger event](https://cloud.ibm.com/docs/openwhisk?topic=cloud-functions-cloud_object_storage#data_structure_trigger_event)

- [x] S3 Trigger event object is the response object from the ListObjects API call. Can we link to the [docs](https://console.bluemix.net/docs/infrastructure/cloud-object-storage-infrastructure/buckets.html#list-objects-in-a-specific-bucket) as a source reference if they need more info on this? 

## [Creating an action to retrieve and process the object](https://cloud.ibm.com/docs/openwhisk?topic=cloud-functions-cloud_object_storage#creating_action_to_retrieve_object) 
 
- [x] Example code needs to have `exports.main = main;` at the end of the file or it won't work when zipped up.

## [Creating an action sequence to retrieve and process the object](https://cloud.ibm.com/docs/openwhisk?topic=cloud-functions-cloud_object_storage#creating-an-action-sequence-to-retrieve-and-process-the-object)

- [x] The instructions do not show how to create the sequence! 
- [x] Why can't the Object Storage package actions be included in the experimental `/whisk.system/cos-experimental` package? 
- [x] Put something a bit more realistic in the demo code below:
```
function main(data) {
  if (data) {
    // Process the object
  }
}
```
- [x] Explain the demo code needs saving in the `myCosAction.js` file

**Comment by csantanapr** [comment]
I would not merge the cos package, cos package users already can install via UI or CLI in their namespace /mynamespace/cloud-object-storage/get-object

Doc should have a sequence that shows how to use the action from the cos package to read the object that the trigger is for and then another action that process the data

**Comment by jthomas** [comment]
@csantanapr Can you provide some background as to why you don't think it should be merged? If people are binding the feed provider package, it seems they will often want to use those actions. What's the downside to pre-installing them?

**Comment by mdeuser** [comment]
thanks for the detailed comments @jthomas! :trophy:   i'm pouring over them now

**Comment by mdeuser** [comment]
@derekpoindexter - i think you've been working on the various doc link breaks...

**Comment by derekpoindexter** [comment]
@mdeuser yep, the links for these should be fixed now: 
    "Provision an instance Object Storage"
    "IBM Cloud Object Storage documentation"

**Comment by csantanapr** [comment]
They can’t be merge even if you want to the trigger lives in system package read only and the cos package is Actions that user choose to install in their namespace. 

Having shared global functions in system has being a pain, people copy them all the time to make small tweaks. We cant update them or improve by the api as their global and not versioned making a change in one action can affect every one. 

Also with my security hat this is code in my app that I can’t scan or control, and the vendor can change at any time braking me or allowing some security bug. 

TLDR; After many years maintaining system shared actions in production is being a big pain for both vendor and users.

**Comment by mdeuser** [comment]
pr https://github.ibm.com/BlueMix-Fabric/bluewhisk-docs/pull/187

**Comment by mdeuser** [comment]
@jthomas regarding
> Can we change the demo code below to be a bit more realistic? Make the console.log print a bit more of a formatted message and return some key fields rather than a null?

the initial code is just a "dummy" action used just to validate the trigger, trigger feed, rule, and action are all connected correctly.  validate is the appearance of the trigger and action activation records

**Comment by derekpoindexter** [comment]
@jthomas here's the latest version of the topic - https://cloud.ibm.com/docs/openwhisk?topic=cloud-functions-pkg_obstorage
